src/kvstore/raftex/SnapshotManager.cpp

```diff
@@ -63,23 +67,31 @@ folly::Future<Status> SnapshotManager::sendSnapshot(std::shared_ptr<RaftPart> pa
                             LOG(INFO) << part->idStr_ << "Finished, totalCount " << totalCount
                                                       << ", totalSize " << totalSize;
                             p.setValue(Status::OK());
+                            onFinished(host);
                         }
                         return;
                     }
                 } catch (const std::exception& e) {
                     LOG(ERROR) << part->idStr_ << "Send snapshot failed, exception " << e.what();
                     p.setValue(Status::Error("Send snapshot failed!"));
+                    onFinished(host);
                     return;
                 }
             }
             LOG(WARNING) << part->idStr_ << "Send snapshot failed!";
             p.setValue(Status::Error("Send snapshot failed"));
+            onFinished(host);
             return;
         });
     });
     return fut;
 }
 
+void SnapshotManager::onFinished(std::shared_ptr<Host> host) {
```

Comment:
modify the host member in snapshotmanager is bad, it is hard understand and the module coupling is complex.

Reply:
We must extend Host's life cycle after the snapshot has been sent out, because the Host may be destructed during this phase